connect/api/src/main/java/org/apache/kafka/connect/data/Values.java

```diff
@@ -713,7 +713,7 @@ protected static String escape(String value) {
         return DOUBLEQOUTE.matcher(replace1).replaceAll("\\\\\"");
     }
 
-    protected static DateFormat dateFormatFor(java.util.Date value) {
+    public static DateFormat dateFormatFor(java.util.Date value) {
```

Comment:
this is a public API change -- everything public in the Connect API is.

do we actually need this here? Scanning through this PR, it looks like it is used in one specific class and then test classes in the same package. Making it package-protected where it is actually used seems sufficient to me, no?

Reply:
package-private is not enough since `Cast` is in package `org.apache.kafka.connect.transforms` tries to use `Values#dateFormatter(java.util.Date)` in `org.apache.kafka.connect.data`.
I think it's fine since we'd want anyone/anything that's manipulating "data values" to use the same formatter, no? so we give access to that method.